proactive-jupyter-kernel/kernel.py

```diff
@@ -961,27 +970,65 @@ def __set_job_name__(self, name):
         self.job_name = name
         return 0
 
-    def __get_result__(self, input_data):
-        job_id = 0
+    def __get_job_id_from_inputs__(self, input_data):
+        if 'job_id' in input_data:
+            return input_data['job_id']
+        if 'job_name' in input_data and input_data['job_name'] != '':
+            if input_data['job_name'] not in self.submitted_jobs_ids:
+                raise ResultError("The job named \'" + input_data['job_name'] + "\' does not exist.")
+            return self.submitted_jobs_ids[input_data['job_name']]
+        raise ParameterError('Invalid parameters and validation step. Please check.')
 
-        if 'id' in input_data:
-            job_id = int(input_data['id'])
-            self.__kernel_print_ok_message__('Getting job ' + str(job_id) + ' output ...\n')
-
-        elif 'name' in input_data and input_data['name'] != '':
-            if input_data['name'] not in self.submitted_jobs_ids:
-                raise ResultError("The job named \'" + input_data['name'] + "\' does not exist.")
-            job_id = self.submitted_jobs_ids[input_data['name']]
-            self.__kernel_print_ok_message__('Getting job \'' + input_data['name'] + '\' output ...\n')
+    def __get_job_result__(self, input_data):
+        job_id = self.__get_job_id_from_inputs__(input_data)
+        self.__kernel_print_ok_message__('Getting job ' + job_id + ' results ...\n')
 
         try:
             job_result = self.gateway.getJobResult(job_id)
         except Exception:
             raise ResultError("Results unreachable for job: " + str(job_id))
 
+        self.__kernel_print_ok_message__('Results:\n')
+        self.__kernel_print_ok_message__(job_result)
+
+    def __get_task_result__(self, input_data):
+        job_id = self.__get_job_id_from_inputs__(input_data)
+        self.__kernel_print_ok_message__('Getting from job ' + job_id + ', task \'' + input_data['task_name']
+                                         + '\' results ...\n')
+
+        try:
+            task_result = self.gateway.getTaskResult(job_id, input_data['task_name'])
+        except Exception:
+            raise ResultError("Results unreachable for job: " + job_id)
+
         self.__kernel_print_ok_message__('Result:\n')
+        self.__kernel_print_ok_message__(str(task_result))
+
+    def __print_job_output__(self, input_data):
+        job_id = self.__get_job_id_from_inputs__(input_data)
+        self.__kernel_print_ok_message__('Getting job ' + job_id + ' console outputs ...\n')
+
+        try:
+            job_result = self.gateway.printJobOutput(job_id)
+        except Exception:
+            raise ResultError("Results unreachable for job: " + str(job_id))
```

Comment:
why str here? no need, and what about job_name with job_id, make more sense right?

Reply:
It's a mistake. I forgot to delete the cast. Thanks!